doc: developer-documentation: Add example how to add a new kselftest (#741)
kernelci/kernelci-pipeline: doc/developer-documentation.md

After following these steps, run your pipeline instance to activate your newly added test configuration.

> ### Example to enable a new kselftest

Reviewer: I'd suggest to make it generic:
```suggestion
### A complete example of enabling a new job
```

Author: Changing it to "A example of enabling a new job"

> The kselftests (found in `<linux_kernel>/tools/testing/selftests`) are enabled suite by suite. Most parameters are already set for other kselftests. Those must be reused while enabling new kselftests which make enabling new tests pretty straight forward. The `dt` kselftest suite can be enabled by:

Reviewer: ```suggestion
Here is an example of enabling `kselftest` (found in `<linux_kernel>/tools/testing/selftests`) job. Most parameters are already set for other kselftests. Those must be reused while enabling new kselftests which makes enabling new tests pretty straight forward. The `dt` kselftest suite can be enabled by:
```

1. Create a job definition in `jobs` section.
```
  kselftest-dt:
    template: kselftest.jinja2
    kind: job

> 

Reviewer: Please drop this empty line.

    params:
      nfsroot: 'http://storage.kernelci.org/images/rootfs/debian/bookworm-kselftest/20240221.0/{debarch}'
      collections: dt
      job_timeout: 10
    rules:
      min_version:
        version: 6
        patchlevel: 7
    kcidb_test_suite: kselftest.dt
```

> 2. Create a `scheduler` entry for `kselftest-dt` job. Lets schedule this suite in `lava-collabora` lab at `kbuild-gcc-12-arm64` build completion event. The entry by using already defined anchors would be:

Reviewer: ```suggestion
2. Create a `scheduler` entry for `kselftest-dt` job. Let's schedule this suite in `lava-collabora` lab at `kbuild-gcc-12-arm64` build completion event. The entry by using already defined anchors would be:
```

Reviewer: This isn't resolved yet.

Author: Seems like there was issue with force push. Updated now.
